sdk/communication/azure-communication-networktraversal/src/main/java/com/azure/communication/networktraversal/CommunicationRelayClientBuilder.java

```diff
@@ -0,0 +1,301 @@
+
+    private final ClientLogger logger = new ClientLogger(CommunicationRelayClientBuilder.class);
+    private String endpoint;
+    private AzureKeyCredential azureKeyCredential;
+    private TokenCredential tokenCredential;
+    private HttpClient httpClient;
+    private HttpLogOptions httpLogOptions = new HttpLogOptions();
+    private HttpPipeline pipeline;
+    private RetryPolicy retryPolicy;
+    private Configuration configuration;
+    private ClientOptions clientOptions;
+    private String connectionString;
+    private final Map<String, String> properties = CoreUtils.getProperties(COMMUNICATION_IDENTITY_PROPERTIES);
+    private final List<HttpPipelinePolicy> customPolicies = new ArrayList<HttpPipelinePolicy>();
+
+    /**
+     * Set endpoint of the service
+     *
+     * @param endpoint url of the service
+     * @return CommunicationRelayClientBuilder
+     */
+    public CommunicationRelayClientBuilder endpoint(String endpoint) {
+        this.endpoint = endpoint;
+        return this;
+    }
+
+    /**
+     * Set endpoint of the service
+     *
+     * @param pipeline HttpPipeline to use, if a pipeline is not
+     * supplied, the credential and httpClient fields must be set
+     * @return CommunicationRelayClientBuilder
+     */
+    public CommunicationRelayClientBuilder pipeline(HttpPipeline pipeline) {
+        this.pipeline = pipeline;
+        return this;
+    }
+
+    /**
+     * Sets the {@link TokenCredential} used to authenticate HTTP requests.
+     *
+     * @param tokenCredential {@link TokenCredential} used to authenticate HTTP requests.
+     * @return The updated {@link CommunicationRelayClientBuilder} object.
+     */
+    public CommunicationRelayClientBuilder credential(TokenCredential tokenCredential) {
+        this.tokenCredential = tokenCredential;
+        return this;
+    }
+
+    /**
+     * Sets the {@link AzureKeyCredential} used to authenticate HTTP requests.
+     *
+    * @param keyCredential The {@link AzureKeyCredential} used to authenticate HTTP requests.
+     * @return The updated {@link CommunicationRelayClientBuilder} object.
+     */
+    public CommunicationRelayClientBuilder credential(AzureKeyCredential keyCredential)  {
+        this.azureKeyCredential = keyCredential;
+        return this;
+    }
+
+    /**
+     * Set endpoint and credential to use
+     *
+     * @param connectionString connection string for setting endpoint and initalizing CommunicationClientCredential
+     * @return CommunicationRelayClientBuilder
+     */
+    public CommunicationRelayClientBuilder connectionString(String connectionString) {
+        CommunicationConnectionString connectionStringObject = new CommunicationConnectionString(connectionString);
+        String endpoint = connectionStringObject.getEndpoint();
+        String accessKey = connectionStringObject.getAccessKey();
+        this
+            .endpoint(endpoint)
+            .credential(new AzureKeyCredential(accessKey));
```

Comment:
This should be done in the build method. Also, don't overwrite the `endpoint` that the user might have already set using the `endpoint()` method above.

Reply:
If they provide a Connection String, they shouldn't provide an endpoint before. That's how we are doing it for other languages for network-traversal and Identity. They can provide Connection String or they set endpoint and credential. So it's ok to override it.